### ISSUE TITLE
Command failed with EACCES

### ISSUE THREAD

**Issue by ammut** [jakebailey/every-ts#58]
I had to `chmod +x node_modules/every-ts/.data/fnm/fnm` before I could `every-ts switch main`. Error message said something about: `spawn /home/samuelv/.nvm/versions/node/v18.18.0/lib/node_modules/every-ts/.data/fnm/fnm EACCES`, not insanely helpful.

**Comment by jakebailey** [comment]
The tool already does that: https://github.com/jakebailey/every-ts/blob/c17074cb9c79e0997ba3114aa24aba7b89bb7798/src/fnm.ts#L64

Do you have a stack trace?

**Comment by ammut** [comment]
Here's what I did:

```console
ammut@local:~/Projects/ts-bug-57861$ rm -rf node_modules
ammut@local:~/Projects/ts-bug-57861$ npm install every-ts

added 32 packages, and audited 33 packages in 616ms

9 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
ammut@local:~/Projects/ts-bug-57861$ every-ts
Cloning TypeScript; this may take a bit...
Cloning into '/home/ammut/Projects/ts-bug-57861/node_modules/every-ts/.data/TypeScript'...
remote: Enumerating objects: 282747, done.
remote: Counting objects: 100% (725/725), done.
remote: Compressing objects: 100% (520/520), done.
remote: Total 282747 (delta 351), reused 413 (delta 160), pack-reused 282022
Receiving objects: 100% (282747/282747), 1001.31 MiB | 10.02 MiB/s, done.
Resolving deltas: 100% (169253/169253), done.
remote: Enumerating objects: 65692, done.
remote: Counting objects: 100% (41260/41260), done.
remote: Compressing objects: 100% (31267/31267), done.
remote: Total 65692 (delta 11010), reused 10242 (delta 9993), pack-reused 24432
Receiving objects: 100% (65692/65692), 28.13 MiB | 9.44 MiB/s, done.
Resolving deltas: 100% (15750/15750), done.
Updating files: 100% (70389/70389), done.
HEAD is now at dc113f8170 Don't indirect through _namespaces in tsserver/typingsInstaller (#57829)
Building TypeScript...
Downloading fnm...
ammut@local:~/Projects/ts-bug-57861$ every-ts switch main
HEAD is now at dc113f8170 Don't indirect through _namespaces in tsserver/typingsInstaller (#57829)
Building TypeScript...
Error: Command failed with EACCES: /home/ammut/Projects/ts-bug-57861/node_modules/every-ts/.data/fnm/fnm install 20
spawn /home/ammut/Projects/ts-bug-57861/node_modules/every-ts/.data/fnm/fnm EACCES
    at ChildProcess._handle.onexit (node:internal/child_process:284:19)
    at onErrorNT (node:internal/child_process:477:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -13,
  code: 'EACCES',
  syscall: 'spawn /home/ammut/Projects/ts-bug-57861/node_modules/every-ts/.data/fnm/fnm',
  path: '/home/ammut/Projects/ts-bug-57861/node_modules/every-ts/.data/fnm/fnm',
  spawnargs: [ 'install', '20' ],
  originalMessage: 'spawn /home/ammut/Projects/ts-bug-57861/node_modules/every-ts/.data/fnm/fnm EACCES',
  shortMessage: 'Command failed with EACCES: /home/ammut/Projects/ts-bug-57861/node_modules/every-ts/.data/fnm/fnm install 20\n' +
    'spawn /home/ammut/Projects/ts-bug-57861/node_modules/every-ts/.data/fnm/fnm EACCES',
  command: '/home/ammut/Projects/ts-bug-57861/node_modules/every-ts/.data/fnm/fnm install 20',
  escapedCommand: '"/home/ammut/Projects/ts-bug-57861/node_modules/every-ts/.data/fnm/fnm" install 20',
  exitCode: undefined,
  signal: undefined,
  signalDescription: undefined,
  stdout: '',
  stderr: '',
  cwd: '/home/ammut/Projects/ts-bug-57861',
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
Unable to build typescript at rev dc113f8170bc9408961e4e89f1ffadf3dff5a41b; please file a bug!
```

So I guess the stacktrace is this:

```stacktrace
Error: Command failed with EACCES: /home/ammut/Projects/ts-bug-57861/node_modules/every-ts/.data/fnm/fnm install 20
spawn /home/ammut/Projects/ts-bug-57861/node_modules/every-ts/.data/fnm/fnm EACCES
    at ChildProcess._handle.onexit (node:internal/child_process:284:19)
    at onErrorNT (node:internal/child_process:477:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
```

**Comment by jakebailey** [comment]
That is utterly confusing. Is there any chance you can do this again and do an `ls -l` in `node_modules/every-ts/.data/fnm` before you would normally do that extra chmod?

**Comment by jakebailey** [comment]
Agh, ok, it just failed for me too.

```
Error: Command failed with EACCES: /home/jabaile/.local/lib/node_modules/every-ts/.data/fnm/fnm install 20
spawn /home/jabaile/.local/lib/node_modules/every-ts/.data/fnm/fnm EACCES
    at ChildProcess._handle.onexit (node:internal/child_process:286:19)
    at onErrorNT (node:internal/child_process:484:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
```

```console
$ ls -l /home/jabaile/.local/lib/node_modules/every-ts/.data/fnm
total 7172
-rw-rw-rw- 1 jabaile jabaile 7340256 Aug 15  2023 fnm
```

I don't even get this.

**Comment by jakebailey** [comment]
My only lead is the fact that this never completes in your trace or mine:

```
HEAD is now at dc113f8170 Don't indirect through _namespaces in tsserver/typingsInstaller (#57829)
Building TypeScript...
Downloading fnm...
ammut@local:~/Projects/ts-bug-57861$ every-ts switch main
```

It just straight up exited. But, in my local checkout, it doesn't. `adm-zip` recently updated so I'm wondering if something broke in that library.

**Comment by jakebailey** [comment]
Yep, this is a bug in adm-zip; rolling back to 0.5.10 fixes it. I'll put out a patch release to pin it back and report that upstream.

**Comment by jakebailey** [comment]
Alright, this should be fixed in 2.0.1, and I've added some testing to CI to help catch this. Thanks for reporting this.
